### ISSUE TITLE
Unmatched result of bitwise_or(56,round(array_min(array_sort("c0")),"c1"))

### ISSUE THREAD

**Issue by kagamiori** [facebookincubator/velox#2795]
```
======> Started iteration 0 (seed: 3891714364)
I1008 10:34:19.652544 1622558 ExpressionFuzzer.cpp:798] Executing expression: bitwise_or(56,round(array_min(array_sort("c0")),"c1"))
I1008 10:34:19.652776 1622558 ExpressionFuzzer.cpp:801] 2 vectors as input:
I1008 10:34:19.652806 1622558 ExpressionFuzzer.cpp:803] 	[CONSTANT ARRAY<TINYINT>: 100 elements, 10 elements starting at 290 {[290->230] -92, [291->280] -92, [292->11] -92, [293->35] -92, [294->247] -92, ...}], [ARRAY ARRAY<TINYINT>: 38 elements, 6 nulls]
I1008 10:34:19.652987 1622558 ExpressionFuzzer.cpp:803] 	[FLAT INTEGER: 100 elements, 13 nulls]
E1008 10:34:19.659252 1622558 Exceptions.h:68] Line: velox/expression/tests/ExpressionFuzzer.cpp:179, Function:compareVectors, Expression: vec1->equalValueAt(vec2.get(), i, i) Different results at idx '0': '-68' vs. 'null', Source: RUNTIME, ErrorCode: INVALID_STATE
terminate called after throwing an instance of 'facebook::velox::VeloxRuntimeError'
  what():  Exception: VeloxRuntimeError
Error Source: RUNTIME
Error Code: INVALID_STATE
Reason: Different results at idx '0': '-68' vs. 'null'
Retriable: False
Expression: vec1->equalValueAt(vec2.get(), i, i)
Function: compareVectors
File: velox/expression/tests/ExpressionFuzzer.cpp
Line: 179
Stack trace:
...
```

**Comment by kagamiori** [comment]
[unmatch_result_repro.zip](https://github.com/facebookincubator/velox/files/9739934/unmatch_result_repro.zip)

**Comment by kgpai** [comment]
FYI - This is again a bug in the simple eval thats used in fuzzer.  I am going to add the propogate nulls by @mbasmanova while I dig into this deeper.

**Comment by kgpai** [comment]
FYI - Still see the bug with the propogate nulls fix, digging deeper.

**Comment by mbasmanova** [comment]
There is a bug in array_sort.

```
  resultElements->copy(
      decodedElements->base(), inputElementRows, /*toSourceRow=*/nullptr);
```

should be something like

```
  resultElements->copy(
      inputElements.get(), inputElementRows, /*toSourceRow=*/nullptr);
```

**Comment by kagamiori** [comment]
Close this issue since it was fixed by #3002 .
